services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/serialization/TransferManagerMarshallingUtils.java

```diff
@@ -0,0 +1,127 @@
+        unmarshallers.put(MarshallingType.NULL, TransferManagerJsonUnmarshaller.NULL);
+        unmarshallers.put(MarshallingType.FLOAT, TransferManagerJsonUnmarshaller.FLOAT);
+        unmarshallers.put(MarshallingType.DOUBLE, TransferManagerJsonUnmarshaller.DOUBLE);
+        unmarshallers.put(MarshallingType.BIG_DECIMAL, TransferManagerJsonUnmarshaller.BIG_DECIMAL);
+        unmarshallers.put(MarshallingType.BOOLEAN, TransferManagerJsonUnmarshaller.BOOLEAN);
+        unmarshallers.put(MarshallingType.SDK_BYTES, TransferManagerJsonUnmarshaller.SDK_BYTES);
+
+        UNMARSHALLERS = Collections.unmodifiableMap(unmarshallers);
+    }
+
+    private TransferManagerMarshallingUtils() {
+    }
+
+    @SuppressWarnings("unchecked")
+    public static <T> TransferManagerJsonMarshaller<T> getMarshaller(T val) {
+        MarshallingType<T> tMarshallingType = toMarshallingType(val);
+        return getMarshaller(tMarshallingType, val);
+    }
+
+    @SuppressWarnings("unchecked")
+    private static <T> MarshallingType<T> toMarshallingType(T val) {
+        MarshallingType<?> marshallingType = MarshallingType.NULL;
+        if (val != null) {
+            for (MarshallingType<?> type : MARSHALLERS.keySet()) {
+                if (type.getTargetClass().isAssignableFrom(val.getClass())) {
+                    marshallingType = type;
+                }
+            }
+        }
+        return (MarshallingType<T>) marshallingType;
+    }
+
+    @SuppressWarnings("unchecked")
+    public static <T> TransferManagerJsonMarshaller<T> getMarshaller(MarshallingType<?> marshallingType, T val) {
+        TransferManagerJsonMarshaller<?> marshaller = MARSHALLERS.get(val == null ? MarshallingType.NULL : marshallingType);
+        if (marshaller == null) {
+            throw new IllegalStateException(String.format("Cannot find a marshaller for marshalling type %s", marshallingType));
+        }
+        return (TransferManagerJsonMarshaller<T>) marshaller;
+    }
+
+    @SuppressWarnings("unchecked")
+    public static TransferManagerJsonUnmarshaller<Object> getUnmarshaller(MarshallingType<?> marshallingType) {
+        TransferManagerJsonUnmarshaller<?> unmarshaller = UNMARSHALLERS.get(marshallingType);
+        if (unmarshaller == null) {
+            throw new IllegalStateException(String.format("Cannot find an unmarshaller for marshalling type %s",
+                                                          marshallingType));
+        }
+        return (TransferManagerJsonUnmarshaller<Object>) unmarshaller;
+    }
+
+    public static SdkField<?> getSdkField(String key) {
+        return GetObjectRequest.builder().build()
+                               .sdkFields().stream()
+                               .filter(f -> f.locationName().equalsIgnoreCase(key))
+                               .findFirst()
+                               .orElseThrow(() -> new IllegalStateException("Could not match a field in GetObjectRequest"));
```

Comment:
This seems a bit inefficient and `GetObjectRequest` has tons of fields. Can we store the sdk fields in a map?